staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -373,24 +373,29 @@ func ValidateCustomResourceDefinitionSubResources(subResources *apiextensions.Cu
 		}
 
 		if len(subResources.Scale.StatusReplicasPath) == 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("scale.statusReplicasPath"), subResources.Scale.StatusReplicasPath, "cannot be empty"))
+			allErrs = append(allErrs, field.Required(fldPath.Child("scale.statusReplicasPath"), ""))
 		} else {
 			// should be constrained json path under .status
 			if errs := validateSimpleJSONPath(subResources.Scale.StatusReplicasPath, fldPath.Child("scale.statusReplicasPath")); len(errs) > 0 {
 				allErrs = append(allErrs, errs...)
 			} else if !strings.HasPrefix(subResources.Scale.StatusReplicasPath, ".status.") {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("scale.specReplicasPath"), subResources.Scale.StatusReplicasPath, "should be a json path under .status"))
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scale.statusReplicasPath"), subResources.Scale.StatusReplicasPath, "should be a json path under .status"))
 			}
 		}
 
-		// if labelSelectorPath is present, it should be a constrained json path.
+		// if labelSelectorPath is present, it should be a constrained json path under .status
 		if len(subResources.Scale.LabelSelectorPath) > 0 {
-			if errs := validateSimpleJSONPath(subResources.Scale.StatusReplicasPath, fldPath.Child("scale.labelSelectorPatch")); len(errs) > 0 {
+			if errs := validateSimpleJSONPath(subResources.Scale.LabelSelectorPath, fldPath.Child("scale.labelSelectorPath")); len(errs) > 0 {
```

Comment:
There was a typo here. We should not use DNSLabel for validation since this fails:

```
CustomResourceDefinition.apiextensions.k8s.io "noxus.mygroup.example.com" is invalid: spec.subResources.scale.labelSelectorPath[1]: Invalid value: "labelSelector": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
```